pycbc/detector.py

```diff
@@ -322,3 +304,96 @@ def overhead_antenna_pattern(right_ascension, declination, polarization):
 def effective_distance(distance, inclination, f_plus, f_cross):
     return distance / np.sqrt( ( 1 + np.cos( inclination )**2 )**2 / 4 * f_plus**2 + np.cos( inclination )**2 * f_cross**2 )
 
+    """ LISA class  """
+
+class LISA(object):
+    def __init__(self, kappa, _lambda_, reference_time = 1126259462.0):
+        self.reference_time = reference_time
+        self.kappa = kappa
+        self._lambda_ = _lambda_
+
+    def get_pos(self, t_gps):
+        if t_gps is None:
+            t_gps = Time(val = self.reference_time, format = 'gps',
+                         scale = 'utc').to_datetime(timezone = None)
+        elif isinstance(t_gps, np.ScalarType):
+            t_gps = Time(val = t_gps, format = 'gps',
+                         scale = 'utc').to_datetime(timezone = None)
+
+        t_gps = np.sum(np.array([t_gps.year - 2034, t_gps.month/12,
+                                 t_gps.day/(4380),
+                                 t_gps.hour/(105120), 
+                                 t_gps.minute/(6307200),
+                                 t_gps.second/(37843200),
+                                 t_gps.microsecond/(37843200*1e-6)]), axis=0)
+
+        n = np.array(range(1, 4))
+        kappa, _lambda_ = 0, 0
+        alpha = 2. * np.pi * t_gps/1 + kappa
+        beta_n = (n - 1) + (2. * np.pi/3) + _lambda_
+        a, L = 1., .1   # units are in AU
+        e = L/(2. * a * np.sqrt(3))
+        _prod_ = a*e*(sin(alpha)*cos(alpha)*sin(beta_n)
+
+        x = a*cos(alpha) + _prod_ - (1 + sin(alpha)**2)*cos(beta_n))
+        y = a*sin(alpha) + _prod_ - (1 + cos(alpha)**2)*sin(beta_n))
+        z = -np.sqrt(3)*a*e*cos(alpha - beta_n)
+        self.location = np.array([x,y,z])
+
+        return self.location
+
+    def plot_orbit(self):
+        from mpl_toolkits.mplot3d import Axes3D
+        import matplotlib.pyplot as plt
+        dec_center = []
+        for i in range(2000):
+          dec_center.append(self.get_pos(self.reference_time + i).mean(axis = 1))
```

Comment:
tried passing GPS time in loop to continuously obtain positions

Reply:
@SSingh087 I don't think we need a method just for the plot. It would be better to have a stand-a-lone script of yours which calls into the pycbc to get the LISADetector object to then make the plot.